### PR TITLE
[Bugfix] Allow decoding complex types like Enums and BaseRecords.

### PR DESCRIPTION
**Pull request by chasdevs**

### CODE REVIEW

File: app/Compiler/Twig/templates/baseRecord.twig [hunk 1]
```diff
@@ -34,14 +40,47 @@ abstract class BaseRecord implements JsonSerializable
...
 
-            if ($prop instanceof ReflectionProperty) {
-                $prop->setAccessible(true);
-                $prop->setValue($this, $value);
+            if (!$prop instanceof ReflectionProperty) {
+                continue;
+            }
+
+            $propName = $prop->getName();
+            if (key_exists($propName, $this->propClassMap)) {
```
Comment (balexander) [inline app/Compiler/Twig/templates/baseRecord.twig:57]: Maybe move `protected $propClassMap = [];` above this function. I get why declaring it above `getValueForComplexPropType` has advantages too, but I was confused when I saw this the first time

File: app/Compiler/Twig/templates/baseRecord.twig [hunk 2]
```diff
@@ -26,6 +28,10 @@ abstract class BaseRecord implements JsonSerializable
         return json_decode(json_encode($mixed), true);
     }
 
+    /**
+     * @param array[string]mixed $array - Array holding arbitrary data to be decoded into this object.
```
Comment (jasonvideoblocks) [inline app/Compiler/Twig/templates/baseRecord.twig]: Does `array[string]mixed` mean something in PHPDoc or is that a typo?

Comment (chasdevs) [reply app/Compiler/Twig/templates/baseRecord.twig]: That was my misguided attempt to type nested arrays from the other day that carried over to this. Going to just keep it as "array"

File: app/Compiler/Twig/templates/baseRecord.twig (hunk 1 above)
Comment (chasdevs) [reply app/Compiler/Twig/templates/baseRecord.twig:57]: Oh yeah forgot I use it up here.

File: app/Compiler/Avro/AvroRecord.php [hunk 3]
```diff
@@ -22,6 +22,9 @@ class AvroRecord implements AvroTypeInterface, AvroNameInterface
     /** @var string[] */
     public $imports = [];
 
+    /** @var array[string]string */
```
Comment (balexander) [inline app/Compiler/Avro/AvroRecord.php]: I think this should be `/** @var string[] */

**Review by balexander — APPROVED** [review]

**Pull request merged**